Wikipedia/Code/MWKImage.m

```diff
@@ -278,4 +278,58 @@ - (BOOL)isLeadImage {
     return [self.article.image isEqualToImage:self];
 }
 
++ (CGSize)minimumImageSizeForGalleryInclusion {
+    return CGSizeMake(80, 80);
+}
+
+- (BOOL)isCanonical {
+    return [MWKImage fileSizePrefix:self.sourceURLString] == NSNotFound;
+}
+
+- (MWKImage *)largestNonCanonicalVariant {
+    NSString *largestNonCanonicalVariantURL = [[[self.article.images imageSizeVariants:self.sourceURLString] bk_select:^BOOL (NSString* url) {
+        return [MWKImage fileSizePrefix:url] != NSNotFound;
+    }] lastObject];
+    return [self.article imageWithURL:largestNonCanonicalVariantURL];
+}
+
+-(UIImage*)imageFromAppImageCache {
```

Comment:
@bgerstle the existing `cachedImageWithURL` returns 'Promise<WMFImageDownload>', which is not what I need here...